### PR TITLE
fix(ftp): Generate member name if missing in `putContents`

### LINKED ISSUE #3305: ZE FTP fails with upload to MVS
**Describe the bug**

ZE FTP fails with upload MVS

**To Reproduce**

1. Go to ZE with ZE FTP ext installed and FTP profile in place
2. Click on FTP profile or PDS to upload a member/PS file
3. See error

**Expected behavior**

Upload to MVS using FTP profile without error/failure

**Screenshots**

Screenshot 2024-11-08 at 11 00 16 AM

**Desktop (please complete the following information):**

- OS: macos
- Zowe Explorer & ZE FTP Version: 3.0.2
- (Optional) Zowe CLI Version:
- (Optional) Are you using Secure Credential Store?

**Additional context**

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

Resolves #3305 

## Release Notes

Milestone: 3.0.3

Changelog:

- Fixed issue where the MVS API `putContents` function did not support PDS members when the member was not specified in the data set name. [#3305](https://github.com/zowe/zowe-explorer-vscode/issues/3305)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [x] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [x] There is coverage for the code that I have added
- [x] I have added new test cases and they are passing
- [x] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

**Review by JillieBeanSim — APPROVED** [review]
LGTM! thanks for the quick fix @traeok

File: packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpMvsApi.ts [hunk 1]
```diff
@@ -128,12 +128,26 @@ export class FtpMvsApi extends AbstractFtpApi implements MainframeInteraction.IM
         const result = this.getDefaultResponse();
         const profile = this.checkedProfile();
 
+        const dsorg = dsAtrribute.apiResponse.items[0]?.dsorg;
+        const isPds = dsorg === "PO" || dsorg === "PO-E";
```
Comment (zFernand0) [inline packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpMvsApi.ts:132]: I'm curious if there are a few other kind of PDS `dsorg`s that we may not be considering. Also, not sure if FTP does support them (like PO-L I think)

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋 
Thanks for the quick fix! 🙏

File: packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpMvsApi.ts (hunk 1 above)
Comment (traeok) [reply packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpMvsApi.ts:132]: That's a good question - we use this check in most places in the codebase, but I think we should try to cover all of the possible PDS `dsorg` values moving forward 😅

**Pull request merged**